src/pygeoprocessing/routing/routing.pyx

```diff
@@ -850,8 +876,12 @@ def fill_pits(
                     xi_q = search_queue.front().xi
                     yi_q = search_queue.front().yi
                     search_queue.pop()
-                    if search_steps > max_pixel_fill_count:
+                    if (max_pixel_fill_count > 0 and

```

Comment:
Okay, so if `max_pixel_fill_count == -1`, we don't want to come in here and quit the search early. We are riding it out until we find that single drain point. (just 'typing' out loud)

Reply:
Well I guess if `single_outlet` is `None`, then we're just searching until we find an actual drain, be it nodata, an edge, or an outlet.

Reply:
Or if it's determined it should be filled.

Reply:
Correct.